src/text-highlighter/text-highlighter.tsx

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+interface Props {
+    highlightedTexts: Array<string>;
+    content: string;
+}
+
+class TextHighLighter extends React.Component<Props, {}> {
+    public render() {
+        const parts = this.props.content.split(new RegExp(`(${this.props.highlightedTexts.join('|')})`, 'g'));
```

Comment:
What happens when you pass a keyword containing `|`. Can you think of any other input that line might be sensitive to?

Reply:
checking on this it currently fails with | cases

Reply:
i have identified the possible list !@#$%^&*()+=\[\]\\';,./{}|":<>?~_-

Reply:
fixed